src/hint_deploy.py

```diff
@@ -363,16 +364,40 @@ def proxy_configure(container, cfg):
         docker_util.exec_safely(container, args)
 
 
+def check_hintr_online(loadbalancer, port, name):
+    result = ""
+    i = 0
+    success_message = (b'{"status":"success",'
+                       b'"errors":null,'
+                       b'"data":"Welcome to hintr"}')
+    while result != success_message and i < 10:
+        try:
+            (_, result) = docker_util.exec_safely(
+                loadbalancer,
+                ["curl", "-s", name + ":" + port]
+            )
+        except Exception as e:
+            print(e)
+            time.sleep(1)
+        i += 1
+    return result == success_message
+
+
 def loadbalancer_register_hintr_api(constellation):
     print("[hintr] Configuring loadbalancer")
     cfg = constellation.data
+    port = str(cfg.hintr_port)
     loadbalancer = constellation.containers.get("hintr", cfg.prefix)
     api_instances = constellation.containers.get("hintr_api", cfg.prefix)
     args = []
     for instance in api_instances:
         args += ["--address", instance.name]
+        is_online = check_hintr_online(loadbalancer, port, instance.name)
+        if not is_online:
+            raise Exception("{} did not come online".format(instance.name))
```

Comment:
if you want to test this easily, then take a deployed version, kill hintr, and run this command. make it easier for yourself by pulling this into a little function that you can call separately with a timeout to it.

```
def ensure_hintr_online(loadbalancer, port, name, timeout=30):
  if not check_hintr_online(loadbalancer, port, name):
    raise Exception(f"{name} did not come online")
```

with your code above being:

```
    for instance in api_instances:
        ensure_online(loadbalancer, port, instance.name)
        args += ["--address", instance.name]

    docker_util.exec_safely(
        loadbalancer, ["configure_backend", "-p", port] + args)
```

and the testing strategy looks like:

```
# kill container, then:
with pytest.raises(Exception, match="did not come online"):
  ensure_online(loadbalancer, port, instance.name, 1)
```